background/services/internal-signer/index.ts

```
@@ -265,21 +273,29 @@ export default class InternalSignerService extends BaseService<Events> {
       return true
     }
 
+    const { vaults, version } = await migrateVaultsToArgon(password)
```

Might be nitpicking here but I think we could improve our approach if we kept the concern for migrating vaults separate from retrieving them. So, rather than returning vaults here, I propose we just return a boolean indicating success or failure and continue using `getEncryptedVaults`.

---

This should be true internally in the encryption layer, but for the caller it's the opposite--encryption layer should be completely in charge of making sure that the data is in the latest format, just like the db objects migrate before returning data.

Vault version is basically a constant across a given run of the extension. All retrievals of encrypted data should automatically reencrypt to the latest version. All encryptions should use the latest version. That fact should be transparent to almost everyone. Migrations should not be even a little bit optional (as skipping a function call might make them). This is how we ensure calling code doesn't make weird decisions about not wanting to upgrade, or weird mistakes about sometimes not doing so, etc: we give no one a choice or a hook. 

To put this in terms of separation of concerns: the encryption layer is concerned with encrypting and decrypting data, and ensuring to the extent possible that all encrypted data is in the latest format.

---

That was my initial approach but then we have to call `getEncryptedVaults` 2/3 times:
1. to get the version and see if it was already migrated, if it wasn't then we can either:
2. pass vaults to migration method or call `getEncryptedVaults` inside the migration method, then
3. to get migrated encrypted vaults in the unlock once again (if we don't want to get them from `migrate...()`

which is not the worst but also seems like a lot of repetition 🤔 will think about it more

---

Thought about this some more. The gap between this and how the internal signer was already implemented is not huge but it's gnarly. I don't think we should deal with it right now, unfortunately.

The way this would manifest is that we would have `getEncryptedVaults` transparently handle vault migration, and `writeLatestEncryptedVault` would be redone as a `writeLatestVault` and would transparently handle encryption and fallbacks.